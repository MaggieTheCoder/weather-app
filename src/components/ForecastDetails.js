import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, humidity, wind } = forecast;
  return (
    <div className="forecast-details" forecast={forecast}>
      <div>
        <h3 className="forecast-details__date">
          {moment(date).format("ddd Do MMM")}
        </h3>
      </div>
      <div
        className="forecast-details__temperature"
        data-testid="forecast-temperature"
      >
        Maximum Temperature: {temperature.max}&deg;C
        <br />
        Minimum Temperature: {temperature.min}&deg;C
      </div>
      <div
        className="forecast-details__humidity"
        data-testid="forecast-humidity"
      >
        Humidity: {humidity}%
      </div>
      <div className="forecast-details__wind" data-testid="forecast-wind">
        Wind: {wind.speed} kmh
        <br />
        Wind direction: {wind.direction}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};

export default ForecastDetails;
