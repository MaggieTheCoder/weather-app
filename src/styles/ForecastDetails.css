.forecast-details {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 10px;
}
