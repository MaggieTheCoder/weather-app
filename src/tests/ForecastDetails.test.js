import React from "react";
import { getByText, render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const forecast = {
    date: 11111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 32,
    wind: {
      speed: 12,
      direction: "e",
    },
  };

  it("renders component correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={forecast} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails forecast={forecast} />
    );

    expect(getByText("Thu 1st Jan")).toHaveClass("forecast-details__date");
    expect(getByTestId("forecast-temperature")).toHaveClass(
      "forecast-details__temperature"
    );
    expect(getByTestId("forecast-humidity")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByTestId("forecast-wind")).toHaveClass("forecast-details__wind");
  });
});
