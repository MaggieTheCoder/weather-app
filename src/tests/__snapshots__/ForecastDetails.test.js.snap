// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders component correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    forecast="[object Object]"
  >
    <div>
      <h3
        class="forecast-details__date"
      >
        Thu 1st Jan
      </h3>
    </div>
    <div
      class="forecast-details__temperature"
      data-testid="forecast-temperature"
    >
      Maximum Temperature: 22°C
      <br />
      Minimum Temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
      data-testid="forecast-humidity"
    >
      Humidity: 32%
    </div>
    <div
      class="forecast-details__wind"
      data-testid="forecast-wind"
    >
      Wind: 12 kmh
      <br />
      Wind direction: e
    </div>
  </div>
</DocumentFragment>
`;
